version: '2.1'

networks:
  myapp_network:
    external: true
    name: myapp_network

services:
  myapp_db:
    image: "percona:latest"
    networks:
      - myapp_network
    environment:
      MYSQL_ROOT_PASSWORD: pass
      MYSQL_DATABASE: myapp_db
    volumes:
      - ./application/mysql/init:/docker-entrypoint-initdb.d
    ports:
      - "3306:3306"
    healthcheck:
      test: ['CMD', 'mysqladmin', '-uroot', '-ppass', 'ping', '-h', '127.0.0.1']
      timeout: 1s
      retries: 30

  vk_api:
    build: ./application/mock/
    networks:
      - myapp_network
    volumes:
      - ./application/mock/flask_app:/app
    working_dir: /app
    entrypoint: ["python", "views.py"]
    ports:
      - "8008:8008"
    depends_on:
      myapp_db:
        condition: service_healthy

  myapp:
    image: "myapp:latest"
    networks:
      - myapp_network
    entrypoint: /app/myapp --config=/config/config.txt
    volumes:
      - ./application/myapp/config:/config
    ports:
      - "8071:8071"
    depends_on:
      myapp_db:
        condition: service_healthy
      vk_api:
        condition: service_started

  myapp_proxy:
    image: "nginx:stable"
    networks:
      - myapp_network
    ports:
      - "8070:8070"
    volumes:
      - ./application/myapp_proxy/nginx/default.conf:/etc/nginx/conf.d/default.conf
    depends_on:
      myapp:
        condition: service_started
